xs/src/libslic3r/GCode/ArcFitting.hpp

```diff
@@ -0,0 +1,43 @@
+#ifndef slic3r_ArcFitting_hpp_
+#define slic3r_ArcFitting_hpp_
+
+#include "libslic3r.h"
+#include "GCode.hpp"
+#include "GCodeReader.hpp"
+
+namespace Slic3r {
+
+enum ArcType {
+    CW,CCW,LINE
+};
+
+class ArcPath : public Polyline {
+    public:
+    Points centers;
+    std::vector<ArcType> arc_types;
+};
+
+std::vector<size_t> arc_detection(Polyline &pl, size_t N, double epsilon);
+std::vector<std::pair<size_t,size_t>> arc_fitting(std::vector<size_t> &arcs, Polyline &pl);
```

Comment:
Some explanation of the expected preconditions/return values here would be helpful.

Reply:
Definitely. It roughly follows the stages outlined in the paper so I hadn't yet commented what they were.